File: B_Sort_the_Array.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    long long elements, L = 1, R = 1, flag = true;
    cin >> elements;
    long long ele[elements + 1];
    ele[0] = 0;
    for (long long i = 1; i <= elements; i++)
        cin >> ele[i];
    for (long long i = 1; i <= elements; i++)
    {
        if (ele[i] >ele[i+1] )
        {
            L = i;
            break;
        }
    }
    for (long long i = elements; i > 0; i--)
    {
        if (ele[i] < ele[i-1])
        {
            R = i;
            break;
        }
    }
    reverse(ele+L,ele+R+1);
    flag=is_sorted(ele,ele+elements+1);
    if(flag)
    {
        cout<<"yes"<<endl<<L<<' '<<R<<endl;
    }
    else cout<<"no"<<endl;
}

File: B_Sort_the_Array.cs
using System;
class program
{
    static void Main()
    {
        bool flag = true;
        int L=0,R=0, n = Convert.ToInt32(Console.ReadLine());
        long[] arr=new long[n];
        string[] str=Console.ReadLine().Split(' ');
        for(int i = 0; i < n; i++)
        {
            arr[i] = Convert.ToInt64(str[i]);
        }
        for(int i = 0; i < arr.Length-1; i++)
        {
            if (arr[i] > arr[i+1])
            {
                L = i;
                break;
            }
        }
        for(int i= arr.Length-1; i>0; i--)
        {
            if (arr[i] < arr[i-1])
            {
                R = i;
                break;
            }
        }
        Array.Reverse(arr,L,R-L+1);
        // foreach(int i in arr)
        //   Console.WriteLine(i);
        for(int i= 0; i < arr.Length-1; i++)
        {
            if (arr[i] > arr[i+1])
            {
                flag = false;
                break;
            }
        }
        if (flag)
        {
            Console.WriteLine("yes");
            Console.WriteLine($"{L+1} {R+1}");
        }
        else
            Console.WriteLine("no");
    }
}